firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -301,6 +329,72 @@ public Query whereArrayContains(@NonNull FieldPath fieldPath, @NonNull Object va
     return whereHelper(fieldPath, Operator.ARRAY_CONTAINS, value);
   }
 
+  /**
+   * Creates and returns a new Query with the additional filter that documents must contain the
+   * specified field, the value must be an array, and that the array must contain any values of the
+   * provided array.
+   *
+   * <p>A Query can have only one whereArrayContainsAny() filter.
```

Comment:
Should we document the other restrictions?  e.g.  "A query can have only one whereArrayContainsAny() filter and it cannot be combined with whereArrayContains() or whereIn()."

And do similar for whereArrayContains() and whereIn().